### ISSUE TITLE
Existing links to the current version's ScalaDoc are broken in 2.12.0

### ISSUE THREAD

**Issue by odd** [scala/scala-lang#515]
Existing links to ScalaDoc that target the current version of Scala (i.e. links such as http://www.scala-lang.org/api/current/index.html#scala.Int) now ends up at the ScalaDoc start page instead of the intended sub page (the sample link must instead now be written as http://www.scala-lang.org/api/current/scala/Int.html). This affects many existing links on StackOverflow.com and other sites with Scala-related information. Perhaps a response with a permanent redirect can be returned for any links using the old format so that the user ends up at the correct page.

(this issue was originally reported as https://issues.scala-lang.org/browse/SI-10029)

**Comment by ashawley** [comment]
> Existing links to ScalaDoc that target the current version of Scala (i.e. links such as http://www.scala-lang.org/api/current/index.html#scala.Int) now ends up at the ScalaDoc start page instead of the intended sub page

Indeed, the new Scaladoc interface is no longer a single-page app supporting both hash fragment links and links to file paths.

> Perhaps a response with a permanent redirect can be returned for any links using the old format so that the user ends up at the correct page.

I don't believe there's a way in HTTP to handle the hash fragment in a URI.  It needs to handled on the client-side by Javascript.  And currently, the anchor links using the hash fragment only support references to the members of classes, and no longer the full namespace and identifier to classes and members in Scaladoc 2.11 and earlier.

The current implementation in 2.12 for hash fragments was provided in scala/scala@1087539

There may be a way to get it corrected so that a failure to match a member, redirects to the class, but that may be perilous.

**Comment by jarrodu** [comment]
I think this issue might be out of scope for this repository. The API docs are not maintained in this Jekyll project. Maybe this should be moved to [https://github.com/scala/bug](https://github.com/scala/bug).

**Comment by SethTisue** [comment]
I don't think this is actionable
